### DISCUSSION TITLE
Minter Problem

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1960] [General]
Can anyone please explain the first minter problem, in Uniswap V2 and how it overcomes the problem?
If possible please give a detailed explanation!!

**Comment by EngrPips** [comment]
I think reading this `https://www.rareskills.io/uniswap-v2-book`  rareskills uniswapV2 book would do more justice to satisfy your curiosity.

**Reply by yasharyan0904** [reply]
I was reading that only :), chapter 6. I has not explained well enough!!

**Reply by yasharyan0904** [reply]
It says this, "Uniswap V2’s defense is to burn first MINIMUM_LIQUIDITY tokens to ensure no-one owns the entire supply of LP tokens and can easily manipulate the price."
The question is why do we need to do that?
Why when liquidity is zero, wee need to multiply 10**3?

**Reply by EngrPips** [reply]
I might need to read the article again myself because it has been long since I read it, and I can't remember every detail, but they don't want a single entity to be able to manipulate the liquidity of the pool such that when other liquidity providers come in, it would affect them.

**Reply by yasharyan0904** [reply]
ook :)

**Comment by yasharyan0904** [comment]
@PatrickAlphaC , if possible can you explain it please !!

**Reply by EngrPips** [reply]
Hello @yasharyan0904, Patrick can be too busy to come aboard, so I will say take a look at this `ethereum stack exchange discussion` `https://ethereum.stackexchange.com/questions/132491/why-minimum-liquidity-is-used-in-dex-like-uniswap` discussion and to gain more in-depth fundamental the first minter issue you can also look at this `https://docs.openzeppelin.com/contracts/4.x/erc4626#inflation-attack`

**Reply by yasharyan0904** [reply]
I know, I just played a bet to know more:)
Thanks a lot, these two cleared my doubt.

**Reply by EngrPips** [reply]
Awesome, I am glad those two resources helped, I need to read them myself.
